### ISSUE TITLE
Introduce number type operations

### ISSUE THREAD

**Issue by steebchen** [prisma/prisma-client-js#655]
SQL allows to atomically operate on number types, included but not limited to incrementing or decrementing a value:

```sql
UPDATE posts SET views = views + 1 WHERE ...
```

This should be possible with Prisma as well. Users are doing [ugly workarounds](https://github.com/prisma/prisma-client-js/issues/349#issuecomment-614315151).

Proposed syntax:

```js
await prisma.user.updateMany({
  where: {},
  data: 
    age: {
      set: 18, // this just sets the value, previous syntax is `data: { age: 18 }`
      // or
      increment: 1, // views = views + 1
      // or
      decrement: 2, // views = views - 2
      // or
      multiply: 3, // views = views * 3
      // or
      divide: 4, // views = views / 4
    },
  },
})
```

**Comment by schickling** [comment]
Related: This is also being addressed in the upcoming client API https://github.com/prisma/specs/issues/356

![image](https://user-images.githubusercontent.com/1567498/79434581-df072780-7fce-11ea-886d-cbc863d099fe.png)

**Comment by cpotemski** [comment]
This is the reason why I don't use prisma atm

**Comment by thebiglabasky** [comment]
Hi,
I went through a first pass shaping the problem and adding some more context in a new issue: https://github.com/prisma/prisma-client-js/issues/775
We will be tracking changes through that one from now on, so feel free to subscribe to it for further updates on the progress.

**Comment by thebiglabasky** [comment]
Hi everyone, since we are now tracking progress on this feature here: https://github.com/prisma/prisma-client-js/issues/775 I will this close this one.

**Comment by steebchen** [comment]
This is now available as a preview feature in [2.6.0](https://github.com/prisma/prisma/releases/tag/2.6.0).
